src/ReferenceList.php

```diff
@@ -241,4 +245,59 @@ public function unserialize( $data ) {
 		$this->__construct( unserialize( $data ) );
 	}
 
+	/**
+	 * Removes duplicates bases on hash value.
+	 *
+	 * @since 0.2
+	 */
+	public function removeDuplicates() {
+		$knownHashes = array();
+
+		/**
+		 * @var Hashable $hashable
+		 */
+		foreach ( iterator_to_array( $this ) as $hashable ) {
+			$hash = $hashable->getHash();
+
+			if ( in_array( $hash, $knownHashes ) ) {
+				$this->detach( $hashable );
+			}
+			else {
+				$knownHashes[] = $hash;
+			}
+		}
+	}
+
+	/**
+	 * The hash is purely valuer based. Order of the elements in the array is not held into account.
+	 *
+	 * @since 0.3
+	 *
+	 * @return string
+	 */
+	public function getValueHash() {
```

Comment:
This is called once in `Statement`. However, it should call the `MapValueHasher` there and this method should be removed as well.